### PR TITLE
User can delete a playlist

### LINKED ISSUE #31: User can delete playlist by id
When a user sends a `DELETE` request to `/api/v1/playlists/:id` it deletes that favorite from the db and returns a status 204 code. If a favorite can't be found, it does nothing to the db and returns a status 404

Example request: `DELETE /api/v1/playlists/1`

Example success response: `status: 204`
Example error response: `status: 404, body: {"error": "No such favorite found. No deletion made."}`

- [x] Happy path 204
- [x] Sad path 404
- [x] Catch 500
- [x] Testing
- [x] Documentation

### PR DESCRIPTION
**Pull request by rhantak**
- Add an action for users to delete a playlist by its id

Closes #31

### CODE REVIEW

File: routes/api/v1/playlists.js [hunk 1]
```diff
@@ -92,8 +92,20 @@ database('playlists')
...
+    .where(request.params)
+    .del()
+    .then(result => {
+      if(result === 1){
+        response.status(204).send()
+      } else if (result === 0) {
+        response.status(404).json({error: "No such playlist found. No deletion made."})
+      }
+    })
+    .catch((error) => response.status(500).json({ error: error }))
```
Comment (ap2322) [inline routes/api/v1/playlists.js]: Let's mask the full error with an error message and console.log the actual error.

File: routes/api/v1/playlists.js [hunk 2]
```diff
@@ -92,8 +92,20 @@ database('playlists')
...
+    .where(request.params)
+    .del()
+    .then(result => {
+      if(result === 1){
+        response.status(204).send()
+      } else if (result === 0) {
+        response.status(404).json({error: "No such playlist found. No deletion made."})
+      }
+    })
+    .catch((error) => response.status(500).json({ error: "Oops, something went wrong!" }))
```
Comment (ap2322) [inline routes/api/v1/playlists.js]: should we also console.log the full error?

File: tests/deletePlaylist.spec.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        .then((playlists) => playlists.length)
+
+      expect(newPlaylistCount).toBe(playlistCount - 1)
+
+      playlistNotFound = await database('playlists').where(playlistOne).select()
+
+      expect(playlistNotFound.length).toBe(0)
+    })
+
+    test('It should send a 404 if no favorite found', async() => {
```
Comment (ap2322) [inline tests/deletePlaylist.spec.js]: copy-pasta favorite to playlists

**Comment by rhantak** [comment]
Ok, made those changes. Thanks!

**Pull request merged**